### DISCUSSION TITLE
Help creating a custom search box

### DISCUSSION THREAD

**Discussion by handshek** [uber/react-map-gl#2398] [Q&A]
I'm trying to create a custom search box with suggestions using the search box API, and my code is turning into spaghetti. Does anyone have any guides or tips on how to implement this? The examples in the mapbox docs aren't very helpful.

**Comment by Pessimistress** [comment]
https://visgl.github.io/react-map-gl/examples/geocoder

**Reply by handshek** [reply]
Hi! Thanks for responding. I did make use of this initially; however, I have a specific requirement that must be met. 
I want the search results to be filtered by the feature type `city`, and unfortunately the geocoder API does not provide this.
Unless I'm missing something?

**Reply by Pessimistress** [reply]
https://github.com/mapbox/mapbox-gl-geocoder/blob/main/API.md#setfilter

**Reply by handshek** [reply]
geocoder API does not have the `city` feature type. Thanks anyway.
